tests/drivers/rtc/rtc_api/src/test_y2k.c

```diff
@@ -10,6 +10,10 @@
 
 #include <time.h>
 
+#if defined(CONFIG_SOC_FAMILY_STM32)
```

Comment:
A more scalable solution here is to add a Kconfig to exclude the Y2K test, and enable this Kconfig in the boards .conf files.

Reply:
I did exclude the y2k test file in CMakeLisks, the `#ifdef` here is just an extra layer to prevent building this test for STM32.